packages/components/src/components/Warning/Warning.tsx

```diff
@@ -24,10 +26,14 @@ export type WarningVariant = Extract<
     'primary' | 'secondary' | 'info' | 'warning' | 'destructive' | 'tertiary'
 >;
 
+const DEFAULT_VARIANT = 'warning' as WarningVariant;
+
 export type WarningProps = AllowedFrameProps & {
     children: ReactNode;
+    extra?: ReactNode;
     className?: string;
     variant?: WarningVariant;
+    rightContent?: ReactNode;
```

Comment:
Isn't string sufficient?

Reply:
It is not because you can add there any component. Ideally predefined ones, but not necessarily.